DRILL-6668: In Web UI, highlight options that are not default values

kkhatua opened the pull request:
This commit introduces a new button on the options page that allows a user to reset an option to its system default value. 

To simplify things, a tooltip is shown when the mouse hovers over the button. If the option value is already default, the button is disabled.
![image](https://user-images.githubusercontent.com/4335237/48571523-dd0dd600-e8bb-11e8-9008-9d5ffc957bec.png)

In addition, a second commit refactor Update button functionality to use same code path and reuses what we already are using to set the default (using AJAX) and auto-refreshing

kkhatua commented:
@paul-rogers since you filed the JIRA, can you review this? Added other reviewers since I couldn't add you.

arina-ielchiieva commented on exec/java-exec/src/main/resources/rest/options.ftl:
```diff
@@ -84,7 +104,9 @@ table.sortable thead .sorting_desc { background-image: url("/static/img/black-de
-                      <button class="btn btn-default" type="submit">Update</button>
+                      <button class="btn btn-default" type="button" onclick="alterSysOptionUsingId('${option.getName()}')">Update</button>
+                      <button class="btn btn-default" type="button" onclick="alterSysOption('${option.getName()}','${option.getDefaultValueAsString()}', '${option.getKind()}')" <#if option.getDefaultValueAsString() == option.getValueAsString()>disabled="true" style="pointer-events:none" <#else>
+                      title="Reset to ${option.getDefaultValueAsString()}"</#if>>Default</button>
```
Maybe better rename button to Reset?

kkhatua replied:
Will do.

arina-ielchiieva commented on exec/java-exec/src/main/resources/rest/options.ftl:
```diff
@@ -21,6 +21,26 @@
+        }
+
+        //Read Values and apply
+        function alterSysOptionUsingId(optionRawName) {
```
Looks like before we just had POST method on the Update button and did not have to do replacements. I would prefer if e stick to previous approach or define one without parsing and replacements.

kkhatua replied:
It does make the changes a little more complex, but the primary motivation was that the same code path is being exercised for making any changes to the options. It's also why I kept it as 2 commits. We can revert it if we don't want a common execution.

arina-ielchiieva replied:
Looks like common execution brings more complexity than should, please revert.

kkhatua replied:
Ok. Will do that and run the tests.

kkhatua commented:
@arina-ielchiieva I've reverted the commit and changed the button label.
However, for some strange reason, the `StatusResourcesTest` kept failing during `LocalFunctionRegistry` initialization.  I had to make some strange changes to the `OptionWrapper` class, and that resolved the issue, and all unit tests passed.

arina-ielchiieva commented:
Overall, looks good.

arina-ielchiieva commented:
+1
